app/scripts/modules/core/src/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx

```diff
@@ -29,10 +34,36 @@ export class ManualJudgmentApproval extends React.Component<
       submitting: false,
       judgmentDecision: null,
       judgmentInput: {},
+      applicationRoles: {},
+      runOnce: true,
+      userRoles: [],
       error: false,
     };
   }
 
+  public componentDidMount() {
+      if (this.state.runOnce) {
+          const applicationName = this.props.execution.application;
+          ApplicationReader.getApplicationPermissions(applicationName)
+              .then(result => {
+                  if (typeof result !== 'undefined') {
+                      this.setState({
+                          applicationRoles: result,
+                          runOnce: false,
+                      })
+                      this.isStageAuthorized();
```

Comment:
This method just returns a boolean without side effects, right? Why are we calling it here?

Reply:
Even if the user does refresh the page, we want to execute this function.